src/kinematics.cpp

```diff
@@ -282,9 +310,14 @@ void Kinematics::update_P() {
     static int counter = 0;
 //    #pragma omp parallel for
     for (int k = 1; k < number_of_links; k++) {
-        Matrix4d transformMatrix_k = world_to_link_transform[k];
-        Matrix3d R_k0 = transformMatrix_k.block(0, 0, 3, 3);
-        for (int a = 1; a <= k; a++) {
+        if(link_names[k] == "head" ||
+            link_names[k] == "upperarm_right" || link_names[k] == "lowerarm_right" || link_names[k] == "hand_right" ||
+            link_names[k] == "upperarm_left" || link_names[k] == "lowerarm_left" || link_names[k] == "hand_left") {
```

Comment:
this list of links should be also defined in a yaml file in the config folder like in the previous comment

you can then , for example, read them into a vector and then your if statement would be smth like
`if (vector.contains(link_names[k])`